### PR TITLE
pi: allow injecting pcap file

### PR DESCRIPTION
**Pull request by lebauce**

### CODE REVIEW

**Comment by lebauce** [comment]
run skydive-ppc64le-tests

File: cmd/client/client.go [hunk 1]
```diff
@@ -62,6 +63,11 @@ func RegisterClientCommands(cmd *cobra.Command) {
 	cmd.AddCommand(EdgeRuleCmd)
 }
 
+func clientError(err error) {
```
Comment (hunchback) [inline cmd/client/client.go]: rather implement func body as:

```
if err != nil {
        logging.GetLogger().Error(err)
        os.Exit(err)
}
```

Comment (safchain) [reply cmd/client/client.go]: So the name of the function is clientError so I guess/hope err is not nil

**Comment by lebauce** [comment]
run skydive-functional-tests-backend-elasticsearch

**Comment by lebauce** [comment]
run skydive-go-fmt

File: flow/probes/pcapsocket.go [hunk 2]
```diff
@@ -33,17 +33,20 @@ import (
...
 // PcapSocketProbe describes a TCP packet listener that inject packets in a flowtable
 type PcapSocketProbe struct {
+	graph     *graph.Graph
 	node      *graph.Node
 	state     int64
 	flowTable *flow.Table
 	listener  *net.TCPListener
 	port      int
 	bpfFilter string
+	injector  *packetinjector.PacketInjector
```
Comment (safchain) [inline flow/probes/pcapsocket.go]: is it used ? or is it a remaining thing ?

Comment (lebauce) [reply flow/probes/pcapsocket.go]: Forgot this one. Removed

File: cmd/client/capture.go [hunk 3]
```diff
@@ -48,6 +49,7 @@ var (
 	ipDefrag           bool
 	reassembleTCP      bool
 	layerKeyMode       string
+	pcapInject         bool
```
Comment (safchain) [inline cmd/client/capture.go]: remaining ?

File: flow/pcap.go [hunk 4]
```diff
@@ -45,12 +45,13 @@ type PcapWriter struct {
...
-	handleRead  *pcapgo.Reader
-	packetsChan chan *PacketSequence
-	bpfFilter   string
+	state          int64
+	replay         bool
+	r              io.ReadCloser
+	handleRead     *pcapgo.Reader
+	packetsChan    chan *PacketSequence
+	bpfFilter      string
+	packetCallback func([]byte)
```
Comment (safchain) [inline flow/pcap.go]: doesn't seem to be used (anymore)

File: cmd/client/capture.go (hunk 3 above)
Comment (lebauce) [reply cmd/client/capture.go]: Removed

File: flow/pcap.go (hunk 4 above)
Comment (lebauce) [reply flow/pcap.go]: Removed

**Pull request merged**